iris-mpc-cpu/src/network/tcp/handle.rs

```diff
@@ -1,331 +0,0 @@
-{
-    fn drop(&mut self) {
-        self.shutdown_ct.cancel();
-        tracing::debug!("TcpNetworkHandle dropped");
-    }
-}
-
-#[async_trait]
-impl<T: NetworkConnection + 'static, C: Client<Output = T> + 'static> NetworkHandle
-    for TcpNetworkHandle<T, C>
-{
-    async fn make_network_sessions(&mut self) -> Result<(Vec<NetworkSession>, CancellationToken)> {
-        let err_ct = CancellationToken::new();
-        // cancel the old token just in case
-        self.connection_state.err_ct().await.cancel();
-        self.connection_state
-            .replace_cancellation_token(err_ct.clone())
-            .await;
-
-        // wait for all peers to establish all connections
-        let mut connections = self
-            .make_peer_connections(self.config.num_connections)
-            .await?;
-
-        connections.sync().await?;
-
-        // calls multiplexer::run() on each TCP/TLS stream
-        let mut tcp_sessions = super::session::make_sessions(
-            connections,
-            self.connection_state.clone(),
-            &self.config,
-            self.next_session_id,
-        )
-        .await;
-
-        self.validate_sessions(&mut tcp_sessions).await?;
-
-        let network_sessions: Vec<_> = tcp_sessions
-            .into_iter()
-            .map(|tcp_session| NetworkSession {
-                session_id: tcp_session.id(),
-                role_assignments: self.role_assignments.clone(),
-                networking: Box::new(tcp_session),
-                own_role: Role::new(self.party_index),
-            })
-            .collect();
-
-        let sessions_per_conn = (0..self.config.num_connections)
-            .map(|idx| self.config.get_sessions_for_connection(idx))
-            .collect_vec();
-        tracing::info!(
-            "make_sessions succeeded. starting id: {} sessions per connection: {:?}",
-            self.next_session_id,
-            sessions_per_conn
-        );
-
-        self.next_session_id = self
-            .next_session_id
-            .saturating_add(self.config.num_sessions);
-        if self.next_session_id >= u32::MAX - self.config.num_sessions {
-            self.next_session_id = 0;
-        }
-
-        Ok((network_sessions, err_ct))
-    }
-
-    async fn make_sessions(&mut self) -> Result<(Vec<Session>, CancellationToken)> {
-        let (network_sessions, ct) = self.make_network_sessions().await?;
-
-        let mut session_futures = vec![];
-        for mut network_session in network_sessions.into_iter() {
-            let fut = async move {
-                let my_session_seed = thread_rng().gen();
-                let prf = setup_replicated_prf(&mut network_session, my_session_seed).await?;
-                Ok::<Session, eyre::Report>(Session {
-                    network_session,
-                    prf,
-                })
-            };
-            session_futures.push(fut);
-        }
-
-        let r = parallelize(session_futures.into_iter()).await?;
```

Comment:
In ampc-common, this was replaced with:

```rust
        let results = join_all(session_futures).await;
        let r = results.into_iter().collect::<Result<Vec<_>>>()?;
```

That is not the same because `parallelize` includes `tokio::spawn`.